packages/wrangler/src/pages/build.ts

```diff
@@ -121,32 +134,110 @@ export const Handler = async ({
 		}
 	}
 
+	// for now let's restrict `outfile` to `_worker.bundle` if the
+	// experimentalWorkerBundle flag is set. Once we remove the flag
+	// this will no longer be an issue
+	outfile = experimentalWorkerBundle ? "_worker.bundle" : outfile;
 	buildOutputDirectory ??= dirname(outfile);
-	try {
-		await buildFunctions({
-			outfile,
-			outputConfigPath,
-			functionsDirectory: directory,
-			minify,
-			sourcemap,
-			fallbackService,
-			watch,
-			plugin,
-			buildOutputDirectory,
-			nodeCompat,
-			routesOutputPath,
+
+	const workerScriptPath = resolvePath(buildOutputDirectory, "_worker.js");
+	const foundWorkerScript = existsSync(workerScriptPath);
+	let bundle: BundleResult | undefined = undefined;
+
+	// `wrangler pages functions build --experimental-worker-bundle=true`
+	if (
+		!isInPagesCI &&
+		experimentalWorkerBundle &&
+		!foundWorkerScript &&
+		directory === "functions" &&
```

Comment:
```suggestion
		!foundWorkerScript &&
```

Any reason _not_ to throw this error in more general circumstances too? I can make it so I'm only calling this when I know a that either `functions/` or `_worker.js` exists.

Reply:
✅